### PR TITLE
Add specs document

### PR DESCRIPTION
**Pull request by xevisalle**
We add an specification document to know exactly how to implement Citadel.

### CODE REVIEW

File: docs/specs/sections/protocol.tex [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	\end{itemize}
+
+	\item (\textbf{user}) $\mathsf{req\_service}$ : Request the service to the SSP, establishing communication using a secure channel, and providing the $\SessionCookie$.
+
+	\item (\textbf{SSP}) $\mathsf{get\_session}$ : Receive a $\Session$ from the list of sessions, where $\Session.\lnullifier = \SessionCookie.\lnullifier$.
+
+	\item (\textbf{SSP}) $\mathsf{grant\_service}$ : Grant or deny the service upon verification of the following steps:
+
+	\begin{itemize}
```
Comment (fed-franz) [inline docs/specs/sections/protocol.tex:57]: remove 'or not' in all points (it's not necessary)

File: docs/specs/sections/protocol.tex [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		\item Check whether or not the values $(\attr, \pk_{\SP}, c)$ included in the $\SessionCookie$ are correct.
+		\item Check whether or not the opening $(\pk_{\mathsf{SSP}}, \mathsf{r})$ included in the $\SessionCookie$ match the $\mathsf{session\_hash}$ found in the $\Session$.
```
Comment (fed-franz) [inline docs/specs/sections/protocol.tex:59]: matches

**Review by fed-franz — CHANGES_REQUESTED** [review]

File: docs/specs/sections/protocol.tex [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+
+		$$\Request = ((\rpk, R), \enc, \nonce)$$
+
+	where 
+
+		$$\enc = \Enc_{\kdh} ((\lpk, R)||\ksym_{\License}; \nonce)$$
+
+
+	\item (\textbf{SP}) $\mathsf{get\_license\_req}$ : Continuously check the network for incoming license requests.
```
Comment (marta-belles) [inline docs/specs/sections/protocol.tex:22]: I would add how this is done, that is, the exact equation with the stealth address that needs to be checked.

File: docs/specs/sections/protocol.tex [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+	\item (\textbf{SP}) $\mathsf{send\_license}$ : Upon receiving the request from a user, define a set of attributes $\attr$ representing the license, and compute a digital signature as follows:
+
+		$$\lsig = \sign_{\sk_{\SP}}(\lpk, \attr)$$
+
+	Then, send the following license to the network:
+
+		$$\License = ((\lpk, R), \enc, \nonce, \pos)$$
```
Comment (marta-belles) [inline docs/specs/sections/protocol.tex:30]: Missing comma at the end of the expression.

File: docs/specs/sections/protocol.tex [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+	where 
+
+		$$\enc = \Enc_{\ksym_{\License}} (\lsig || \attr; \nonce)$$
+
+	\item (\textbf{user}) $\mathsf{get\_license}$ : Receive the license by scanning the incoming transactions.
```
Comment (marta-belles) [inline docs/specs/sections/protocol.tex:36]: Same comment as before, which formula exactly should be checked here?

File: docs/specs/images/circuit_prove_nft.svg
Comment (marta-belles) [inline docs/specs/images/circuit_prove_nft.svg:1]: I suggest to replace the "0" by "true" to avoid any confusion.

**Review by marta-belles — CHANGES_REQUESTED** [review]

File: docs/specs/sections/definitions.tex [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+
+    \item \textbf{License:} A license is an asset that represents the right of a user to use a given service. The structure is as follows:
+
+    \begin{center}
+        \begin{tabular}{ |c|c|c|c| } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $(\lpk, R)$ & StealthAddress & It is a license stealth address of the user. \\
```
[the remark this answers is not in the document]
Comment (xevisalle) [reply docs/specs/sections/definitions.tex:34]: yes

File: docs/specs/sections/definitions.tex [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        $\enc$ & PoseidonCipher & It is a ciphertext of size 4. \\
+        $\nonce$ & BlsScalar & It is a randomness needed to compute $\enc$. \\ 
+        $\pos$ & BlsScalar & It is the position of the license into a Merkle tree of licenses. \\
```
[the remark this answers is not in the document]
Comment (xevisalle) [reply docs/specs/sections/definitions.tex:37]: I'll add a new section for storage of everything once we have agreed with the other on how this will work.

File: docs/specs/sections/definitions.tex [hunk 8]
```diff
@@ -0,0 +1,99 @@
+\subsection{The Roles involved} 
+
```
[the remark this answers is not in the document]
Comment (xevisalle) [reply docs/specs/sections/definitions.tex:2]: sure, that can be a follow up version of the specs, along with the storage section

File: docs/specs/sections/overview.tex [hunk 9]
```diff
@@ -0,0 +1,18 @@
+
+\subsection{What is Citadel} 
+
+A Self-Sovereign Identity (SSI) protocol serves the purpose of allowing users of a given service to manage their identities in a fully transparent manner. In other words, every user can know which information about them is shared with other parties, and accept or deny any request for personal information.
+
+\verb!Citadel! is a SSI protocol build on top of Dusk Network. Users of a service can get a \textit{license}, which represents their \textit{right} to use such a service. In particular, \verb!Citadel! allows for the following properties:
+
+\begin{itemize}
+	\item \textbf{Proof of Ownership:} a user of a service is able to prove ownership of a license that allows them to use such a service.
```
[the remark this answers is not in the document]
Comment (miloszm) [reply docs/specs/sections/overview.tex]: I'd suggest:
"user of a service is able to prove ownership of a license which allows her or him to use a service"

File: docs/specs/sections/overview.tex [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+	\item \textbf{Proof of Validity:} users can prove ownership of a valid license, that has not been revoked.
+ 	\item \textbf{Unlinkability:} no one can link any activity with other activities done in the network.
+ 	\item \textbf{Decentralized Nullification:} when a user spends a license, everyone in the network learns that this happened, so it cannot be spent again.
```
[the remark this answers is not in the document]
Comment (miloszm) [reply docs/specs/sections/overview.tex]: I'd replace `activities done in the network` with `activities in the network`
also, instead of `learns that this happened`, I'd rather write `learns that this has happened`

File: docs/specs/sections/overview.tex [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+\subsection{What is Citadel} 
+
+A Self-Sovereign Identity (SSI) protocol serves the purpose of allowing users of a given service to manage their identities in a fully transparent manner. In other words, every user can know which information about them is shared with other parties, and accept or deny any request for personal information.
+
+\verb!Citadel! is a SSI protocol build on top of Dusk Network. Users of a service can get a \textit{license}, which represents their \textit{right} to use such a service. In particular, \verb!Citadel! allows for the following properties:
+
+\begin{itemize}
+	\item \textbf{Proof of Ownership:} a user of a service is able to prove ownership of a license that allows them to use such a service.
+	\item \textbf{Proof of Validity:} users can prove ownership of a valid license, that has not been revoked.
+ 	\item \textbf{Unlinkability:} no one can link any activity with other activities done in the network.
```
[the remark this answers is not in the document]
Comment (miloszm) [reply docs/specs/sections/overview.tex:11]: I'd convert all instances of `activities done in the network` to `activities in the network`.

File: docs/specs/sections/overview.tex [hunk 12]
```diff
@@ -0,0 +1,18 @@
+
+\subsection{What is Citadel} 
+
+A Self-Sovereign Identity (SSI) protocol serves the purpose of allowing users of a given service to manage their identities in a fully transparent manner. In other words, every user can know which information about them is shared with other parties, and accept or deny any request for personal information.
+
+\verb!Citadel! is a SSI protocol build on top of Dusk Network. Users of a service can get a \textit{license}, which represents their \textit{right} to use such a service. In particular, \verb!Citadel! allows for the following properties:
+
+\begin{itemize}
+	\item \textbf{Proof of Ownership:} a user of a service is able to prove ownership of a license that allows them to use such a service.
+	\item \textbf{Proof of Validity:} users can prove ownership of a valid license, that has not been revoked.
```
[the remark this answers is not in the document]
Comment (miloszm) [reply docs/specs/sections/overview.tex]: I have similar doubts - does valid mean both `ownership` and `being not revoked` or just `being not revoked` ?

File: docs/specs/sections/definitions.tex [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+\begin{itemize}
+    \item \textbf{Request:} A request includes the encryption of a stealth address belonging to the user, where the license has to be sent to, and a symmetric key. The structure is as follows:
+
+    \begin{center}
+        \begin{tabular}{ |c|c|c|c| } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $(\rpk, R)$ & StealthAddress & It is a request stealth address of the SP. \\
```
[the remark this answers is not in the document]
Comment (miloszm) [reply docs/specs/sections/definitions.tex:19]: I think `request stealth address` is confusing - if it is a user's stealth address encrypted in some way, I'd write `encrypted stealth address of the user` or something like that, but not use the noun `request` as an adjective.

File: docs/specs/sections/definitions.tex (hunk 6 above)
[the remark this answers is not in the document]
Comment (miloszm) [reply docs/specs/sections/definitions.tex:34]: Does `spend` mean something idempotent, like proving ownership only, or it is the act of final nullification? Because the word `spend` intuitively suggests that after that the license is `spent`, no longer usable.

File: docs/specs/sections/definitions.tex [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+        $\mathsf{session\_hash}$ & BlsScalar & The hash of the public key of the SSP together with some randomness. \\ 
+        $\lnullifier$ & BlsScalar & The nullifier of a given license. \\ 
+        $\com_0^{hash}$ & BlsScalar & A hash commitment of the public key of the SP. \\ 
+        $\com_1$ & JubJubExtended & A Pedersen Commitment of the attributes. \\ 
+        $\com_2$ & JubJubExtended & A Pedersen Commitment of the $c$ value. \\ 
+        \hline
+        \end{tabular}
+    \end{center}
+
+    \item \textbf{SessionCookie:} a session cookie is a secret value known only by the user and the SSP. It contains a set of openings to a given set of commitments. The structure is as follows:
```
[the remark this answers is not in the document]
Comment (xevisalle) [reply docs/specs/sections/definitions.tex:79]: Anyone knowing the user can pretend to be the owner of the license, thus using such a session.

File: docs/specs/sections/definitions.tex (hunk 6 above)
[the remark this answers is not in the document]
Comment (xevisalle) [reply docs/specs/sections/definitions.tex:34]: that's the case. You usually spend.... an """"instantiation""""" of a license. If it can be used many times, you will be able to instatiate it multiple times.

File: docs/specs/images/circuit_prove_nft.svg
Comment (zaira-pindado) [inline docs/specs/images/circuit_prove_nft.svg:1]: I don't clearly see the uniqueness of the nullifiers. The nullifier is computed by the hash() of lpk' and c. As I see it, c is not unique by definition and lpk' can not be unique. 
The comparison to Phoenix is clear, there the nullifier is computed by the hash of nsk' (analogous to lpk' here) and the position in the Merkle tree. The position is something unique, so since the hash is deterministic we always have a different nullifier (independently of nsk').

Comment (xevisalle) [reply docs/specs/images/circuit_prove_nft.svg:1]: If the user doesn't follow the protocol and asks for licenses reusing `lpk`, once he will spend one of those licenses, he will nullify all the others, as the network won't accept the same nullifier twice. 

On the other hand, we could think of an adversary seeing `lpk` in the network, and reusing it to nullify others' licenses. BUT, as the adversary cannot compute `lpk'` because he doesn't know `lsk`, he won't be able to nullify such a license.

Comment (xevisalle) [reply docs/specs/images/circuit_prove_nft.svg:1]: hence, it is not a problem. Plus, no one can compute `sig.session_hash` but the user.

Comment (zaira-pindado) [reply docs/specs/images/circuit_prove_nft.svg:1]: Ok, I see the difference respect to Phoenix in the fact that here it is the user itself who creates these keys. then, there's no problem in not having the requirement of uniqueness since it affects negatively to the user itself.
What's the role of c in the nullification process then? maybe lpk' is enough?

File: docs/specs/sections/protocol.tex [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+\begin{enumerate}
+	\item (\textbf{user}) $\mathsf{send\_license\_req}$ : compute a license stealth address $(\lpk, R)$ belonging to the user, using the user's own public key, and also an additional key $\ksym_{\License} = \hb(\lsk)G$, by computing first the user's $\lsk$. Then, compute the request stealth address $(\rpk, R)$ and $\kdh$ using the SP's public key. And finally send the following request to the network:
+
+
+		$$\Request = ((\rpk, R), \enc, \nonce)$$,
+
+	where 
+
+		$$\enc = \Enc_{\kdh} ((\lpk, R)||\ksym_{\License}; \nonce)$$.
```
Comment (fed-franz) [inline docs/specs/sections/protocol.tex:19]: I think the `R` for `lpk` is different from the `R` for `rpk`. 
It should be `R'` or sth

File: docs/specs/sections/protocol.tex [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+
+		$$\License = ((\lpk, R), \enc, \nonce, \pos)$$,
+
+	where 
+
+		$$\enc = \Enc_{\ksym_{\License}} (\lsig || \attr; \nonce)$$.
+
+	\item (\textbf{user}) $\mathsf{get\_license}$ : receive the license by scanning the incoming transactions.
+
+	\item (\textbf{user}) $\mathsf{call\_nullify\_license}$ : when desiring to use the license, nullify it by executing a call to the license contract. The following steps are performed:
```
Comment (fed-franz) [inline docs/specs/sections/protocol.tex:38]: I think this should be a separate workflow.
One workflow is for getting a license and another one is to use it.

File: docs/specs/images/circuit_prove_nft.svg
Comment (zaira-pindado) [reply docs/specs/images/circuit_prove_nft.svg:1]: ok, makes sense! thank you

File: docs/specs/sections/definitions.tex (hunk 13 above)
[the remark this answers is not in the document]
Comment (marta-belles) [reply docs/specs/sections/definitions.tex:19]: I am still confused here... So, we have two stealth addresses:
- (rpk, R): request stealth address of the SP -> is this stealth address used by the SP to identify which requests are sent to him?
- (lpk, R): license stealth address of the user -> is this stealth address used by the user to identify which licenses are sent to him?

File: docs/specs/sections/definitions.tex [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+
+    \begin{center}
+        \begin{tabular}{ |c|c|c|c| } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $\lpk$ & JubJubAffine & The license public key.\\ 
+        $\lpk'$ & JubJubAffine & The license public key prime.\\ 
+        $\lsig$ & Signature & The signature of the license. \\ 
+        $\com_0^{hash}$ & BlsScalar & A hash commitment of the public key of the SP. \\
```
[the remark this answers is not in the document]
Comment (marta-belles) [reply docs/specs/sections/definitions.tex:52]: I see your point but I think the wording may be a bit confusing. Could we maybe simply call it hash or commitment?

**Review by marta-belles — CHANGES_REQUESTED** [review]
Some of the previous comments were not resolved, I pointed back again to those and added a couple more.

File: docs/specs/sections/protocol.tex [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+
+	\begin{itemize}
+		\item The user issues a transaction that calls the license contract, which includes a ZKP that is computed out of the gadget depicted in Figure \ref{fig:circuit_prove_nft}.
+		\item The network validators will execute the smart contract, which verifies the proof. Upon success, the following session will be added to a shared list of sessions:
```
Comment (fed-franz) [inline docs/specs/sections/protocol.tex:42]: Where is this list stored? In the license contract?

File: docs/specs/sections/protocol.tex [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+	\item (\textbf{user}) $\mathsf{req\_service}$ : request the service to the SSP, establishing communication using a secure channel, and providing the $\SessionCookie$.
+
+	\item (\textbf{SSP}) $\mathsf{get\_session}$ : receive a $\Session$ from the list of sessions, where $\Session.\lnullifier = \SessionCookie.\lnullifier$.
+
+	\item (\textbf{SSP}) $\mathsf{grant\_service}$ : grant or deny the service upon verification of the following steps:
+
+	\begin{itemize}
+		\item Check whether the values $(\attr, \pk_{\SP}, c)$ included in the $\SessionCookie$ are correct.
+		\item Check whether the opening $(\pk_{\mathsf{SSP}}, \mathsf{r})$ included in the $\SessionCookie$ matches the $\mathsf{session\_hash}$ found in the $\Session$.
+		\item Check whether the openings $((\pk_{\SP}, \mathsf{s_0}), (\attr, \mathsf{s_1}), (c, \mathsf{s_2}))$ included in the $\SessionCookie$ match the commitments ($\com_0^{hash}, \com_1, \com_2$) found in the $\Session$.
```
Comment (fed-franz) [inline docs/specs/sections/protocol.tex:60]: `$\SessionCookie$` is never defined and appears as `sc` in the text, so it's hard to understand what it is

File: docs/specs/sections/protocol.tex [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+	\centering
+	\setlength{\fboxsep}{5pt}%
+	\setlength{\fboxrule}{0.3pt}%
+	\fbox{
+		\includegraphics[width=460pt,draft=false]{images/circuit_prove_nft.eps}}
+	\caption{Arithmetic circuit for proving a license's ownership.}
+	\label{fig:circuit_prove_nft}
+\end{figure}
+
+Furthermore, the SSP might request the user to nullify the license they are using (e.g. this is a single-use license, like entering a concert). This is done through the computation of $\lnullifier$. The deployment of this part of the circuit has two different possibilities:
```
Comment (fed-franz) [inline docs/specs/sections/protocol.tex:75]: Is this related to the workflow or the circuit? It seems like they are separate things

**Review by fed-franz — DISMISSED** [review]

File: docs/specs/main.tex [hunk 21]
```diff
@@ -0,0 +1,185 @@
+\documentclass[10pt]{article}
```
Comment (miloszm) [inline docs/specs/main.tex:1]: As discussed, let's unify method names, change the following names:
send_license_req --> request_license,
get_license_req --> get_license_request,
send_license --> issue_license,
call_nullify_license --> use_license,
req_service --> request_service

**Review by miloszm — CHANGES_REQUESTED** [review]

File: docs/specs/sections/protocol.tex (hunk 20 above)
Comment (xevisalle) [reply docs/specs/sections/protocol.tex:75]: it's more related to how the application CAN work

File: docs/specs/sections/protocol.tex (hunk 18 above)
Comment (xevisalle) [reply docs/specs/sections/protocol.tex:42]: yes!

File: docs/specs/sections/protocol.tex [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+	where 
+
+		$$\enc = \Enc_{\ksym_{\License}} (\lsig || \attr; \nonce)$$.
+
+	\item (\textbf{user}) $\mathsf{get\_license}$ : receive the license by scanning the incoming transactions.
+
+	\item (\textbf{user}) $\mathsf{call\_nullify\_license}$ : when desiring to use the license, nullify it by executing a call to the license contract. The following steps are performed:
+
+	\begin{itemize}
+		\item The user issues a transaction that calls the license contract, which includes a ZKP that is computed out of the gadget depicted in Figure \ref{fig:circuit_prove_nft}.
```
[the remark this answers is not in the document]
Comment (xevisalle) [reply docs/specs/sections/protocol.tex:41]: nowhere yet, that should be the next section to write

File: docs/specs/sections/definitions.tex [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+\begin{itemize}
+    \item \textbf{Request:} a request includes the encryption of a stealth address belonging to the user, where the license has to be sent to, and a symmetric key. The structure is as follows:
+
+    \begin{center}
+        \begin{tabular}{ |c|c|c|c| } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $(\rpk, R)$ & StealthAddress & It is a request stealth address of the SP. \\
+        $\enc$ & PoseidonCipher & It is a ciphertext of size 6. \\
```
[the remark this answers is not in the document]
Comment (xevisalle) [reply docs/specs/sections/definitions.tex:20]: well, the size can be put in the type column

File: docs/specs/sections/protocol.tex (hunk 16 above)
Comment (xevisalle) [reply docs/specs/sections/protocol.tex:38]: We will create another section to depict the workflows at the code level. I'd stick with it for a general overview.

File: docs/specs/sections/definitions.tex [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+
+\begin{itemize}
+    \item \textbf{Request:} a request includes the encryption of a stealth address belonging to the user, where the license has to be sent to, and a symmetric key. The structure is as follows:
+
+    \begin{center}
+        \begin{tabular}{ | p{0.15\linewidth} | p{0.2\linewidth} | p{0.55\linewidth} | } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $(\rpk, R_{\req})$ & StealthAddress & It is a request stealth address of the LP. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:19]: It is a request stealth address ~~of~~ **for** the LP.

File: docs/specs/sections/definitions.tex [hunk 25]
```diff
@@ -0,0 +1,98 @@
...
+        $\enc$ & PoseidonCipher[6] & It is the encryption of a license stealth address of the user and a symmetric key. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:20]: [...] license stealth address ~~of~~ **for** the user [...]

File: docs/specs/sections/definitions.tex (hunk 24 above)
Comment (marta-belles) [reply docs/specs/sections/definitions.tex:19]: @xevisalle, maybe you were gramatically right with the use of the proposition "of", but I suggest using "for" because it captures the idea that the LP does not have a stealth address, but it is created by someone else _for_ the LP to use it.

File: docs/specs/sections/definitions.tex [hunk 26]
```diff
@@ -0,0 +1,98 @@
...
+
+    \item \textbf{License:} a license is an asset that represents the right of a user to use a given service. The structure is as follows:
+
+    \begin{center}
+        \begin{tabular}{ | p{0.15\linewidth} | p{0.2\linewidth} | p{0.55\linewidth} | } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $(\lpk, R_{\lic})$ & StealthAddress & It is a license stealth address of the user. \\
+        $\enc$ & PoseidonCipher[4] & It is the encryption of some user attributes and its signature. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:34]: By "its signature" do you mean the user's signature? If so, it should be _It is the encryption of some user attributes and ~~its~~ **their** signature".

File: docs/specs/sections/definitions.tex [hunk 27]
```diff
@@ -0,0 +1,98 @@
...
+    \begin{center}
+        \begin{tabular}{ | p{0.15\linewidth} | p{0.2\linewidth} | p{0.55\linewidth} | } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $\lpk$ & JubJubAffine & The license public key of the user.\\ 
+        $\lpk'$ & JubJubAffine & A variation of the license public key of the user computed with a different generator.\\ 
+        $\lsig$ & Signature & The signature of the license attributes. \\ 
+        $\com_0^{hash}$ & BlsScalar & A hash of the public key of the LP. \\ 
+        $\com_1$ & JubJubExtended & A Pedersen Commitment of the attributes. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:52]: Pedersen Commitment --> Pedersen **c**ommitment

File: docs/specs/sections/definitions.tex [hunk 28]
```diff
@@ -0,0 +1,98 @@
...
+        \begin{tabular}{ | p{0.15\linewidth} | p{0.2\linewidth} | p{0.55\linewidth} | } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $\lpk$ & JubJubAffine & The license public key of the user.\\ 
+        $\lpk'$ & JubJubAffine & A variation of the license public key of the user computed with a different generator.\\ 
+        $\lsig$ & Signature & The signature of the license attributes. \\ 
+        $\com_0^{hash}$ & BlsScalar & A hash of the public key of the LP. \\ 
+        $\com_1$ & JubJubExtended & A Pedersen Commitment of the attributes. \\ 
+        $\com_2$ & JubJubExtended & A Pedersen Commitment of the $c$ value. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:53]: Pedersen Commitment --> Pedersen **c**ommitment

File: docs/specs/sections/definitions.tex [hunk 29]
```diff
@@ -0,0 +1,98 @@
...
+
+    \begin{center}
+        \begin{tabular}{ | p{0.15\linewidth} | p{0.2\linewidth} | p{0.55\linewidth} | } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $\mathsf{session\_hash}$ & BlsScalar & The hash of the public key of the SP together with some randomness. \\ 
+        $\sessionid$ & BlsScalar & The id of a session open using a given license. \\ 
+        $\com_0^{hash}$ & BlsScalar & A hash of the public key of the LP. \\ 
+        $\com_1$ & JubJubExtended & A Pedersen Commitment of the attributes. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:72]: Pedersen Commitment --> Pedersen **c**ommitment

File: docs/specs/sections/definitions.tex [hunk 30]
```diff
@@ -0,0 +1,98 @@
...
+    \begin{center}
+        \begin{tabular}{ | p{0.15\linewidth} | p{0.2\linewidth} | p{0.55\linewidth} | } 
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $\mathsf{session\_hash}$ & BlsScalar & The hash of the public key of the SP together with some randomness. \\ 
+        $\sessionid$ & BlsScalar & The id of a session open using a given license. \\ 
+        $\com_0^{hash}$ & BlsScalar & A hash of the public key of the LP. \\ 
+        $\com_1$ & JubJubExtended & A Pedersen Commitment of the attributes. \\ 
+        $\com_2$ & JubJubExtended & A Pedersen Commitment of the $c$ value. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:73]: Pedersen Commitment --> Pedersen **c**ommitment

File: docs/specs/sections/definitions.tex [hunk 31]
```diff
@@ -0,0 +1,98 @@
...
+        \hline
+        \textbf{Element} & \textbf{Type} & \textbf{Info.} \\
+        \hline
+        $\pk_{\mathsf{SP}}$ & JubJubAffine & The public key of the SP. \\
+        $r_\mathsf{session}$ & BlsScalar & Randomness for computing the session hash. \\
+        $\sessionid$ & BlsScalar & The id of a session open using a given license. \\ 
+        $\pk_{\LP}$ & JubJubAffine & The public key of the LP. \\ 
+        $\attr$ & JubJubScalar & The attributes of the user. \\ 
+        $c$ & JubJubScalar & The challenge value. \\ 
+        $\mathsf{s_0}$ & JubJubScalar & Randomness used to compute the $\com_0^{hash}$. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:91]: Randomness used to compute ~~the~~ [...].

File: docs/specs/sections/definitions.tex [hunk 32]
```diff
@@ -0,0 +1,98 @@
...
+        $\mathsf{s_1}$ & BlsScalar & Randomness used to compute the $\com_1$. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:92]: Randomness used to compute ~~the~~ [...].

File: docs/specs/sections/definitions.tex [hunk 33]
```diff
@@ -0,0 +1,98 @@
...
+        $\mathsf{s_2}$ & BlsScalar & Randomness used to compute the $\com_2$. \\
```
Comment (marta-belles) [inline docs/specs/sections/definitions.tex:93]: Randomness used to compute ~~the~~ [...].

File: docs/specs/sections/protocol.tex [hunk 34]
```diff
@@ -0,0 +1,105 @@
+
+In Citadel, each party involved in the protocol keeps static keys, as we detail now. Let $G, G'\gets\mathbb{J}$ be two generators for the subgroup $\mathbb{J}$ of order $t$ of the Jubjub. The keys of each party as the following.
```
Comment (marta-belles) [inline docs/specs/sections/protocol.tex:2]: [...] of order $t$ of the Jubjub **elliptic curve**.

Comment (marta-belles) [inline docs/specs/sections/protocol.tex:2]: [...] of each party ~~as~~ **are** the following.

File: docs/specs/sections/protocol.tex [hunk 35]
```diff
@@ -0,0 +1,105 @@
...
+\begin{itemize}%[label=$\bullet$]
+	\item \textit{Secret key:} $\sk = (a,b)$, where $a,b\gets \F_t$.
+	\item \textit{Public key:} $\pk = (A,B)$, where $A = a G$ and $B = b G$.
+\end{itemize}
+
+The workflow of the Citadel protocol is depicted in Figure \ref{fig:protocol}, and described with full details as follows.
+
+\begin{figure}[h]
+	\centering
+		\includegraphics[width=390pt,draft=false]{images/protocol.eps}
```
Comment (marta-belles) [inline docs/specs/sections/protocol.tex:13]: I think that the figure should reflect that LP and SP can be different entities, and point the arrows to the right entity.

File: docs/specs/sections/protocol.tex [hunk 36]
```diff
@@ -0,0 +1,105 @@
...
+	where 
+
+		$$\enc = \Enc_{\kreq} ((\lpk, R_{\lic})||\klic; \nonce).$$
+
+
+	\item (\textbf{LP}) $\mathsf{get\_license\_request}$ : continuously check the network for incoming license requests, by checking if $\rpk \stackrel{?}{=} \hb(\tkreq) G + B_{\LP}$, where $\tkreq = a_{\LP}R_{\req}$.
+
+	\item (\textbf{LP}) $\mathsf{issue\_license}$ : upon receiving a request from a user, define a set of attributes $\attr$ representing the license, and compute a digital signature as follows:
+
+		$$\lsig = \sign_{\sk_{\SP}}(\lpk, \attr)$$.
```
Comment (marta-belles) [inline docs/specs/sections/protocol.tex:50]: You missed an end dot after the double dollar, it should be inside;)

File: docs/specs/sections/protocol.tex [hunk 37]
```diff
@@ -0,0 +1,105 @@
...
+
+\begin{figure}[h]
+	\centering
+		\includegraphics[width=390pt,draft=false]{images/protocol.eps}
+	\caption{Overview of the protocol messages exchanged between the user, the Dusk Network, and the SP.}
+	\label{fig:protocol}
+\end{figure}
+
+\begin{enumerate}
+	\item (\textbf{user}) $\mathsf{request\_license}$ : compute a license stealth address $(\lpk, R_{\lic})$ belonging to the user, using the user's own public key, as follows.
```
Comment (marta-belles) [inline docs/specs/sections/protocol.tex:19]: I am missing the computation of `lpk'` in the protocol. Should it be computed here? 

I am particularly interested because in one of my last comments you said that the SP has the opening of the session_id, so they can check if `c` is correct, and I would like to understand how lpk' is computed and why SP has access to it.

**Review by marta-belles — CHANGES_REQUESTED** [review]

File: docs/specs/sections/definitions.tex (hunk 26 above)
Comment (xevisalle) [reply docs/specs/sections/definitions.tex:34]: no, the signature of the attributes

File: docs/specs/sections/protocol.tex (hunk 37 above)
Comment (xevisalle) [reply docs/specs/sections/protocol.tex:19]: the SP has the opening to com_3, to check if the value `c` was correctly set.

File: docs/specs/sections/protocol.tex (hunk 35 above)
Comment (xevisalle) [reply docs/specs/sections/protocol.tex:13]: As we discussed, in a future version of the specs we will include how the user interacts with the contract and will make this distinction.

**Review by marta-belles — APPROVED** [review]
